### PR TITLE
Problem: goreleaser config not releasing for Arm64 (fix #376)

### LINKED ISSUE #376: Problem: goreleaser config not releasing for Arm64
Darwin arm64 for M1
Linux arm64 for AWS Graviton etc.

### PR DESCRIPTION
**Pull request by leejw51crypto**
Solution: upgrade golang to 1.16.1
to compile for apple m1

### CODE REVIEW

**Review by tomtau — CHANGES_REQUESTED** [review]
the goreleaser here isn't still configured to release arm64? or would you want to split it into 2 PRs (upgrade go toolchain and update goreleaser settings)?

**Comment by leejw51crypto** [comment]
yes, i'll add build step for apple m1.

**Comment by leejw51crypto** [comment]
## osxcross links 
**osxcross/target/bin/**
1. **ls -la arm64-apple-darwin20.2-clang**
arm64-apple-darwin20.2-clang -> x86_64-apple-darwin20.2-wrapper
2. **ls -la x86_64-apple-darwin20.2-clang**
x86_64-apple-darwin20.2-clang -> x86_64-apple-darwin20.2-wrapper
3. **ls -la o64-clang**
o64-clang -> x86_64-apple-darwin20.2-wrapper

File: .goreleasermac.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+before:
```
Comment (tomtau) [inline .goreleasermac.yml:1]: is it necessary to have this duplicate config? why not just ".goreleaser.yml"

File: .github/workflows/goreleasemac.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+name: goreleasermac
```
Comment (tomtau) [inline .github/workflows/goreleasemac.yml:1]: why this separate pipeline? can't one just extend "make release-dry-run" and "make release" used in https://github.com/crypto-org-chain/chain-main/blob/master/.github/workflows/goreleaser.yml ?

**Review by tomtau — CHANGES_REQUESTED** [review]
what about Linux arm64?

File: .github/workflows/goreleasemac.yml (hunk 2 above)
Comment (leejw51crypto) [reply .github/workflows/goreleasemac.yml:1]: troian/golang-cross doesn't support arm64+ darwin, yet
```
    goos:
      - darwin
    goarch:
      - arm64
      
```

Comment (leejw51crypto) [reply .github/workflows/goreleasemac.yml:1]: error is  like this
image

Comment (tomtau) [reply .github/workflows/goreleasemac.yml:1]: It supports Linux-arm64 though -- so perhaps, the easiest solution may be just to extend the current goreleaser config for Linux-arm64 (so one can deploy e.g. on AWS Graviton https://pages.awscloud.com/rs/112-TZM-766/images/2020_0501-CMP_Slide-Deck.pdf ) ... for M1, there are no inexpensive "servers" one can easily deploy to, so not having official binaries isn't a big problem -- for development, one can build them themselves or use Rosetta.
Later, when the "Darwin-arm64" is supported on troian/golang-cross, we can just extend the config

Comment (leejw51crypto) [reply .github/workflows/goreleasemac.yml:1]: adding to current yml is possible, i'll include it.

File: .goreleasermac.yml (hunk 1 above)
Comment (leejw51crypto) [reply .goreleasermac.yml:1]: i'll remove .gorelasemac.yml, move it to current yml.

Comment (leejw51crypto) [reply .goreleasermac.yml:1]: linux arm64 is possible, i'll add it.

File: .github/workflows/goreleasemac.yml (hunk 2 above)
Comment (leejw51crypto) [reply .github/workflows/goreleasemac.yml:1]: yes, i'll add linux arm64, that would be done very easily.
apple m1 binary is tested successfully, i'll submit to this pr

**Comment by leejw51crypto** [comment]
linux arm64 not working correctly, fixing now

**Comment by leejw51crypto** [comment]
linux arm64( arm8), mac m1 all working
 
for linux arm64
      - CGO_ENABLED=1
      - CC=x86_64-w64-mingw32-gcc
      - CXX=x86_64-w64-mingw32-g++

**Review by tomtau — APPROVED** [review]

**Comment by leejw51crypto** [comment]
image

**Comment by leejw51crypto** [comment]
temporarily applied #406, and built in other repo

**Pull request merged**